spec/jobs/in_person/enrollments_ready_for_status_check/batch_processor_spec.rb

```diff
@@ -0,0 +1,229 @@
+        expect do
+          batch_processor.process_batch(messages, analytics_stats)
+        end.to raise_error(error)
+        expect(analytics_stats).to eq(expected_analytics_stats)
+      end
+
+      it 'invalid item is marked as processed and deleted' do
+        expect(enrollment_pipeline).to receive(:process_message).
+          with(message).and_return(false).once
+        expect(sqs_batch_wrapper).to receive(:delete_message_batch).
+          with(messages).and_return(delete_result).once
+        expect(delete_result).to receive(:failed).and_return([])
+        expect(delete_result).to receive(:successful).and_return(messages)
+        expect(error_reporter).not_to receive(:report_error)
+        expected_analytics_stats = {
+          **analytics_stats,
+          fetched_items: 1,
+          invalid_items: 1,
+          deleted_items: 1,
+          processed_items: 1,
+        }
+        batch_processor.process_batch(messages, analytics_stats)
+        expect(analytics_stats).to eq(expected_analytics_stats)
+      end
+
+      it 'valid item is marked as processed and deleted' do
+        expect(enrollment_pipeline).to receive(:process_message).
+          with(message).and_return(true).once
+        expect(sqs_batch_wrapper).to receive(:delete_message_batch).
+          with(messages).and_return(delete_result).once
+        expect(delete_result).to receive(:failed).and_return([])
+        expect(delete_result).to receive(:successful).and_return(
+          [
+            successful_delete,
+          ],
+        )
+        expect(error_reporter).not_to receive(:report_error)
+        expected_analytics_stats = {
+          **analytics_stats,
+          fetched_items: 1,
+          valid_items: 1,
+          deleted_items: 1,
+          processed_items: 1,
+        }
+        batch_processor.process_batch(messages, analytics_stats)
+        expect(analytics_stats).to eq(expected_analytics_stats)
+      end
+
+      it 'item is marked as processed but fails to be deleted' do
+        expect(enrollment_pipeline).to receive(:process_message).
+          with(message).and_return(true).once
+        expect(sqs_batch_wrapper).to receive(:delete_message_batch).
+          with(messages).and_return(delete_result).once
+        error_entry = failed_delete
+        expect(delete_result).to receive(:failed).and_return(
+          [
+            error_entry,
+          ],
+        )
+        error_entry_hash = {
+          id: 123,
+        }
+        expect(error_entry).to receive(:to_h).and_return(error_entry_hash)
+        expect(delete_result).to receive(:successful).and_return([])
+        expect(error_reporter).to receive(:report_error).with(
+          'Failed to delete item from queue',
+          sqs_delete_error: error_entry_hash,
+        ).once
+        expected_analytics_stats = {
+          **analytics_stats,
+          fetched_items: 1,
+          valid_items: 1,
+          deleted_items: 0,
+          processed_items: 1,
```

Comment:
The default `analytics_stats` hash has all of the items as zero. The overrides here are primarily intended to represent changes in the items. I have set some of the zero redundantly because it matters for understanding the individual test's behavior.